### PR TITLE
Add ruff, black & isort configs

### PR DESCRIPTION
**Pull request by MetRonnie**
I often find myself typing out some code that wouldn't pass flake8, then hitting "format selection" or "organise imports" in VSCode. I am using [isort](https://pycqa.github.io/isort/index.html) and [ruff](https://docs.astral.sh/ruff/) for this (having briefly used [black](https://black.readthedocs.io/en/stable/index.html) before ruff).

I think it would be useful to define our configs for these tools on a project level so we can agree the style and avoid unnecessary diff resulting from this going forwards.

I've chosen defaults that I think best match our current prevailing style. If we want to haggle over this, make it brief!

To illustrate, I've ruffened/blackened and isorted `pathutil.py` as a small example. 

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes
- [x] No tests, changelog, docs needed
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -22,3 +22,27 @@ showcontent = true
...
+[tool.black]
+line-length = 79
+target-version = ['py37']
+skip-string-normalization = true
+
+
+[tool.isort]
+profile = "black"
+line_length = 79
+force_grid_wrap = 4
```
Comment (oliver-sanders) [inline pyproject.toml:46]: [Haggle] Would personally prefer a lower value. I use 2.

File: cylc/flow/pathutil.py [hunk 2]
```diff
@@ -17,16 +17,25 @@
 
 import errno
 import os
-from pathlib import Path
 import re
+from pathlib import Path
 from shutil import rmtree
```
Comment (oliver-sanders) [inline cylc/flow/pathutil.py:22]: This diff goes against project style (sorts **p**athlib after **r**e), but there's a config to change this if we would like to stick with the status quo.

File: pyproject.toml [hunk 3]
```diff
@@ -22,3 +22,27 @@ showcontent = true
...
+lines_after_imports = 2
+combine_as_imports = true
```
[the remark this answers is not in the document]
Comment (wxtim) [reply pyproject.toml:48]: I agree with that.

[the remark this answers is not in the document]
Comment (MetRonnie) [reply pyproject.toml:48]: I prefer the `^import`s being first, but not too fussed

**Review by oliver-sanders — APPROVED** [review]
Should be uncontroversial as it's basically just documenting what we do already.

We should probably sync this change to other repos in time :shrug: (I too have my own local `.isort` config to allow me to autoformat code selections in my test editor).

**Review by wxtim — APPROVED** [review]

**Pull request merged**